app/src/main/java/com/into/websoso/ui/createFeed/CreateFeedViewModel.kt

```diff
@@ -55,19 +69,50 @@ class CreateFeedViewModel
                 isSpoiled.value = feed.isSpoiler
                 isPublic.value = feed.isPublic
                 _categories.addAll(createCategories(feed.feedCategory))
+                if (feed.imageUrls.isNotEmpty()) loadFeedImages(feed.feedId)
             } ?: _categories.addAll(createCategories())
 
             isActivated.addSource(content) { updateIsActivated() }
         }
 
+        private fun loadFeedImages(feedId: Long) {
+            viewModelScope.launch {
+                val result = loadExistImages(feedId)
+
+                result.onSuccess { uris ->
+                    if (uris.isNotEmpty()) {
+                        addCompressedImages(uris)
+                    }
+                }
+            }
+        }
+
+        private suspend fun loadExistImages(feedId: Long) =
+            runCatching {
+                val feed = feedRepository.fetchFeed(feedId)
+                val downloadedUris = mutableListOf<Uri>()
+
+                for (url in feed.images) {
+                    runCatching {
+                        feedRepository.downloadImage(url).getOrThrow()
+                    }.getOrNull()?.let { uri ->
+                        downloadedUris.add(uri)
+                    }
+                }
```

Comment:
r: 코루틴을 활용해서 이미지를 병렬적으로 로딩하면 좋을 것 같아유

```suggestion
 feed.images.map { url ->
            async {
                runCatching {
                    feedRepository.downloadImage(url).getOrThrow()
                }.onFailure {
                    Log.w("ImageDownload", "Failed to download: $url", it)
                }.getOrNull()
            }
        }.awaitAll().filterNotNull()
```

Reply:
> r: 코루틴을 활용해서 이미지를 병렬적으로 로딩하면 좋을 것 같아유

afdcb2f

Reply:
뷰모델스코프로 바꿔주십시오